changelog/2751.bugfix

```diff
@@ -0,0 +1 @@
+Fix Pytest's testdir fixture may end up creating flaky tests.
```

Comment:
@tarcisiofischer if you have the time, could you please change this to:

```
``testdir`` now uses use the same method used by ``tmpdir`` to create its temporary directory. This changes the final structure of the ``testdir`` directory slightly, but should not affect usage in normal scenarios and avoids a number of potential problems.
```

Reply:
No problem 👍